lib/matplotlib/animation.py

```diff
@@ -554,7 +554,38 @@ def isAvailable(cls):
             return False
         return True
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, loop=0, **kwargs):
+        '''PillowWriter
+
+        Parameters
+        ----------
+        fps: int
+            Framerate for the gif.
+        loop: int
+            The number of times that the gif will loop.
+            A value of 0 is endless.
+        codec: string or None, optional
+            The codec to use. If ``None`` (the default) the ``animation.codec``
+            rcParam is used.
+        bitrate: int or None, optional
+            The bitrate for the saved movie file, which is one way to control
+            the output file size and quality. The default value is ``None``,
+            which uses the ``animation.bitrate`` rcParam.  A value of -1
+            implies that the bitrate should be determined automatically by the
+            underlying utility.
+        extra_args: list of strings or None, optional
+            A list of extra string arguments to be passed to the underlying
+            movie utility. The default is ``None``, which passes the additional
+            arguments in the ``animation.extra_args`` rcParam.
+        metadata: Dict[str, str] or None
+            A dictionary of keys and values for metadata to include in the
+            output file. Some keys that may be of use include:
+            title, artist, genre, subject, copyright, srcform, comment.
+        '''
+        if type(loop) == int:
```

Comment:
bools are ints. So ```not isinstance(True, int)``` will return false. Which will result in passing a value of ```True``` aka ```1``` to loop which won't loop, but one would expect loop=True to loop. See my option 3 for a way to implement support for both bools and ints, but this will force just ```ints```.

Reply:
Though I suppose
```python
if type(loop) != int:
    raise ValueError(...)
self.loop=loop
```
But it only saves one of code, and I don't really feel like pushing another commit to this unless I have too 😛.